pkg/resources/management.cattle.io/v3/cluster/validator.go

```diff
@@ -107,7 +107,12 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 	}
 
 	if request.Operation == admissionv1.Create || request.Operation == admissionv1.Update {
-		// no need to validate the local cluster, or imported cluster which represents a KEv2 cluster (GKE/EKS/AKS) or v1 Provisioning Cluster
+		if fieldErr := common.CheckCreatorIDAndNoCreatorRBAC(newCluster); fieldErr != nil {
+			return admission.ResponseBadRequest(fieldErr.Error()), nil
+		}
+
```

Comment:
I think this should go before or after [L95](https://github.com/rancher/webhook/pull/511/files/1d2da326e32215e9baefa122eea248ce3425e547..6bd7ee78378c33405ecfe7769d40deae81f2d8b1#diff-d7e4efad65a61bc47176fc39ab03add1805d4effff6cade04a89c7eec05c36d7R95)
IMHO Creator annotation (any) checks only make sense for Create operation and this case only in local/upstream cluster.

Reply:
IMHO similar to creatorId and creator-principal-name, no-creator-rbac annotation should be immutable. The only allowed update should be removing the annotation. 
So we should add it ot this check https://github.com/rancher/webhook/blob/6b74a9a9006ccb27e74146a91899a02dd03a7d3b/pkg/resources/common/validation.go#L101